src/main.c

```diff
@@ -142,10 +147,160 @@ static bool is_valid_stream(char *s)
 	return true;
 }
 
+char* get_saved_file_path(void)
+{
+	const char* xdg_state_home = getenv("XDG_STATE_HOME");
+	const char* home_path = getenv("HOME");
```

Comment:
I don't think there is a need to getenv() both and then later have the `if`.
I would call the variable `home_path` and get `XDG_STATE_HOME`.
Then I'd null check and if its null Id do the getenv on HOME.